classes/userGroup/relationships/UserGroupStage.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * @file classes/userGroup/relationships/UserUserGroup.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class \PKP\userGroup\UserGroup
+ * @ingroup userGroup
+ *
+ * @see DAO
+ *
+ * @brief UserGroup metadata class.
+ */
+
+namespace PKP\userGroup\relationships;
+
+use PKP\facades\Locale;
+use PKP\db\DAORegistry;
+use PKP\user\User;
+use PKP\userGroup\UserGroup;
+use APP\facades\Repo;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+
+class UserGroupStage extends \Illuminate\Database\Eloquent\Model
+{
+    //
+    // The following attributes configure Eloquent's expectations for this model
+    //
+    public $timestamps = false;
+    public $incrementing = false;
+    protected $primaryKey = null;
+    protected $fillable = ['userGroupId', 'stageId', 'contextId'];
+
+    /**
+     * The table associated with the model.
+     *
+     * @var string
+     */
+    protected $table = 'user_group_stage';
+
+    public function userGroup(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value, $attributes) => Repo::userGroup()->get($attributes['user_group_id']),
+            set: fn($value) => $value->getId()
+        );
+    }
+
+    //
+    // The following two functions permit lowerCamelCase accessors in place of snake_case column names
+    // (e.g. $userUserGroup->userGroupId rather than $userUserGroup->user_group_id)
+    //
+    public function stageId(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes['stage_id'],
+            set: fn($value) => ['stage_id' => $value]
+        );
+    }
+
+    public function userGroupId(): Attribute
+    {
+        return Attribute::make(
+            get: fn($userGroup, $attributes) => $attributes['user_group_id'],
+            set: fn($value) => ['user_group_id' => $value]
+        );
+    }
+
+    public function contextId(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes['context_id'],
+            set: fn($value) => ['context_id' => $value]
+        );
+    }
+
+    //
+    // The following two functions allow for queries to be executed that don't require the database column
+    // names to be exposed in calling code (as ->where would do). Called e.g.: UserUserGroup::withUserId(123)
```

Comment:
Likewise, these comments can be removed.

Reply:
Removed from 
- `UserGroupStage`
- `UserUserGroup`